test/integration/reload_test.lua

```diff
@@ -59,7 +62,90 @@ local function reload_roles(srv)
     t.assert_equals({ok, err}, {true, nil})
 end
 
+local task_sleep_on_10_eval = [[
+    for i = 1,100 do
+        box.space.customers:insert({i})
+    end
+    local expirationd = require('expirationd')
+    local tuples_cnt = 0
+    local is_expired_sleep = function()
+        tuples_cnt = tuples_cnt + 1
+        if tuples_cnt == 10 then
+            require("fiber").sleep(60)
+        end
+        return true
+    end
+    task = expirationd.start("walk_all", box.space.customers.id, is_expired_sleep,
+                             {process_expired_tuple = function() return true end,
+                              force = true})
+
+    for i = 1,100 do
```

Comment:
It works fine ok because we have a 60-seconds sleep inside is_expired call.

We need to wait just a very short time to iterate over 10 tuples in the memory.

Reply:
try:
```
--- a/test/integration/reload_test.lua
+++ b/test/integration/reload_test.lua
@@ -82,12 +82,6 @@ local task_sleep_on_10_eval = [[
                              {process_expired_tuple = function() return true end,
                               force = true})
 
-    for i = 1,100 do
-        if tuples_cnt >= 10 then
-            break
-        end
-        fiber.yield()
-    end
 ]]
 
 local task_first_tuple_eval = [[
@@ -104,13 +98,6 @@ local task_first_tuple_eval = [[
     task = expirationd.start("walk_all", box.space.customers.id, is_expired_tuple,
                              {force = true})
 
-    for i = 1,100 do
-        if tuple ~= nil then
-            break
-        end
-        fiber.yield()
-    end
-
     task:kill()
 
     return tuple or {}
@@ -118,6 +105,8 @@ local task_first_tuple_eval = [[
 
 function g.test_task_continue_after_reload(cg)
     cg.cluster:server('s1-master').net_box:eval(task_sleep_on_10_eval)
+    local tuple = cg.cluster:server('s1-master').net_box:eval(task_first_tuple_eval)
+    t.assert_equals(tuple, {10})
 
     reload_roles(cg.cluster:server('s1-master'))
 
@@ -129,6 +118,8 @@ function g.test_task_continue_after_stop_and_reload(cg)
     cg.cluster:server('s1-master').net_box:eval(task_sleep_on_10_eval .. [[
         task:stop()
     ]])
+    local tuple = cg.cluster:server('s1-master').net_box:eval(task_first_tuple_eval)
+    t.assert_equals(tuple, {10})
 
     reload_roles(cg.cluster:server('s1-master'))
 
@@ -140,6 +131,8 @@ function g.test_task_not_continue_after_kill_and_reload(cg)
     cg.cluster:server('s1-master').net_box:eval(task_sleep_on_10_eval .. [[
         task:kill()
     ]])
+    local tuple = cg.cluster:server('s1-master').net_box:eval(task_first_tuple_eval)
+    t.assert_equals(tuple, {10})
 
     reload_roles(cg.cluster:server('s1-master'))
```

Reply:
It will be fragile because ```local tuple = cg.cluster:server('s1-master').net_box:eval(task_first_tuple_eval)``` actually starts a new task and it can change the state.

Reply:
I still don't get why we can't use the same approach as in unit test.

Reply:
We have nothing to wait for at the test instance of Tarantool.

At the cluster instance I can replace `fiber.yield()` by `helpers.repeat()` inside evals if you want. It does not look like a critical change. Done:

https://github.com/tarantool/expirationd/blob/cc7605aba3dfd2820716f65facb4456b29af7681/test/integration/reload_test.lua#L87-L92

Reply:
Oh, yeah, now I get what the struggle was. I think it should be okay now.